src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc

```diff
@@ -164,139 +190,195 @@ const std::string LightName(const smart_objects::SmartObject& light_name) {
   return ok ? name : "unknown";
 }
 
-bool CheckLightDataByCapabilities(
+ModuleCapability GetItemCapability(
     const smart_objects::SmartObject& capabilities,
-    const smart_objects::SmartObject& light_data) {
+    const std::map<std::string, std::string>& mapping,
+    const std::string& request_parameter,
+    const mobile_apis::Result::eType& switched_off_result) {
+  const auto it = mapping.find(request_parameter);
+
+  if (it == mapping.end()) {
+    LOG4CXX_DEBUG(logger_,
+                  "Parameter " << request_parameter
+                               << " doesn't exist in capabilities.");
+    return std::make_pair("", capabilitiesStatus::missedParam);
+  }
+
+  const std::string& caps_key = it->second;
+
+  LOG4CXX_DEBUG(logger_,
+                "Checking request parameter "
+                    << request_parameter
+                    << " with capabilities. Appropriate key is " << caps_key);
+
+  if (!capabilities.keyExists(caps_key)) {
+    LOG4CXX_DEBUG(logger_,
+                  "Capability " << caps_key
+                                << " is missed in RemoteControl capabilities");
+    return std::make_pair("", capabilitiesStatus::missedParam);
+  }
+
+  if (!capabilities[caps_key].asBool()) {
+    LOG4CXX_DEBUG(logger_,
+                  "Capability "
+                      << caps_key
+                      << " is switched off in RemoteControl capabilities");
+    ModuleCapability status =
+        std::make_pair("", capabilitiesStatus::missedParam);
+    if (mobile_apis::Result::READ_ONLY == switched_off_result) {
+      status = std::make_pair("", capabilitiesStatus::readOnly);
+    }
+    return status;
+  }
+
+  return std::make_pair("", capabilitiesStatus::success);
+}
+
+ModuleCapability GetLightDataCapabilities(
```

Comment:
No header or description for some of these new methods? I am having a hard time understanding the specific roles for GetLightDataCapabilities,  GetLightNameCapabilities, GetItemCapability, GetControlDataCapabilities, GetHmiControlDataCapabilities, and GetModuleDataCapabilities?